wpilibj/src/main/java/edu/wpi/first/wpilibj/logging/LogFile.java

```diff
@@ -0,0 +1,179 @@
+import java.util.Locale;
+
+/**
+ * A LogFile writes text to log in a file.
+ */
+public class LogFile extends Writer {
+  private final String m_filePrefix;
+  private final String m_fileExtension;
+  private OutputStream m_file;
+  private Date m_time;
+  private int m_timeIntervalBeforeRenaming = 86400;
+
+  /**
+   * Instantiate a LogFile passing in its prefix and its extension.
+   *
+   * <p>If you want the file to be saved in a existing directory, you can add its
+   * path before the file prefix. Exemple : to save the file in a usb stick on
+   * the roborio ("/media/sda1/") : LogFile("/media/sda1/log").
+   *
+   * @param filePrefix The prefix of the LogFile.
+   * @param fileExtension The extension of the LogFile (without dot).
+   */
+  public LogFile(String filePrefix, String fileExtension) {
+    m_filePrefix = filePrefix;
+    m_fileExtension = fileExtension;
+
+    m_time = new Date(0);
+    String fileName = createFilename(m_time);
+    try {
+      m_file = Files.newOutputStream(Paths.get(fileName));
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  /**
+   * Instantiate a LogFile passing in its prefix.
+   *
+   * <p>If you want the file to be saved in a existing directory, you can add its
+   * path before the file prefix. Exemple : to save the file in a usb stick on
+   * the roborio ("/media/sda1/") : LogFile("/media/sda1/log").
+   *
+   * @param filePrefix The prefix of the LogFile.
+   */
+  public LogFile(String filePrefix) {
+    this(filePrefix, "txt");
+  }
+
+  /**
+   * Instantiate a LogFile.
+   */
+  public LogFile() {
+    this("log", "txt");
+  }
+
+  /**
+   * Write text in the LogFile.
+   *
+   * @param text The text to be logged in the file.
+   */
+  public void log(String text) {
+    try {
+      m_file.write(text.getBytes());
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+    updateFilename();
+  }
+
+  /**
+   * Write text in the LogFile and add a new line.
+   *
+   * @param text The text to be logged in the file.
+   */
+  public void logln(String text) {
+    log(text + "\n");
+  }
+
+  /**
+   * Get the name the file.
+   *
+   * @return The name of the file.
+   */
+  public String getFileName() {
+    return createFilename(m_time);
+  }
+
+  /**
+   * Set the time interval after which the file will be renamed in seconds.
+   *
+   * @param seconds The time interval after which the file will be renamed in seconds.
+   */
+  public void setTimeIntervalBeforeRenaming(int seconds) {
+    m_timeIntervalBeforeRenaming = seconds;
+  }
+
+  /**
+   * Write text in the LogFile.
+   *
+   * @param text The array of chars to write from
+   * @param off The index into the array to start writing from
+   * @param len The number of chars to write
+   */
+  @Override
+  public void write(char[] text, int off, int len) throws IOException {
+    char[] textCopy = new char[len];
+    System.arraycopy(text, off, textCopy, 0, len);
```

Comment:
This doesn't seem right... why would this need to make a copy?

Reply:
You were right I changed it in https://github.com/wpilibsuite/allwpilib/pull/1820/commits/7e96ff53c547e7ded465b67277c2e3d26274ab49

Reply:
That still makes a copy?

Reply:
In fact it certainly does. But I don't see other ways to do it ...